code/game/gamemodes/miniantags/guardian/types/lightning.dm

```diff
@@ -18,6 +18,15 @@
 	var/list/enemychains = list()
 	var/successfulshocks = 0
 
+/mob/living/simple_animal/hostile/guardian/beam/Life(seconds, times_fired) //Dies if the summoner dies
+	..()
+	if(summoner)
+		if(!(NO_SHOCK in summoner.mutations))
+			summoner.mutations.Add(NO_SHOCK)
```

Comment:
This probably shouldn't be running in Life, and instead be a one time thing. I was thinking you could add a new proc, or even just add code in `New()` for guardians (though that would need modifications to `New()`'s args). The code would be roughly what you have here, except it only needs to run once.

For example with New:

```dm
// Add the user var to the New proc for this example
/mob/living/simple_animal/hostile/guardian/beam/New(loc, mob/living/user)
    if(!user)
        return ..()
    summoner = user
    if(!(NO_SHOCK in summoner.mutations))
		summoner.mutations.Add(NO_SHOCK)
```
Also you may want to remove the shock immunity from the host when the guardian dies, something like:
```dm
/mob/living/simple_animal/hostile/guardian/beam/death(gibbed)
    if(summoner && (NO_SHOCK in summoner.mutations))
		summoner.mutations.Remove(NO_SHOCK)
    return ..()
```

Reply:
Sure, one moment

Reply:
don't think I need user coded like that though, since I can just check for if it has a summoner?

Reply:
You would with the New route, because when you get a guardian, the code does this:
```
	var/mob/living/simple_animal/hostile/guardian/G = new pickedtype(user)
	G.summoner = user
```
Alternative would be to create another proc, and then run it after this line above

Reply:
Getting indentation errors with that code, along with bad argument definition

Reply:
Oh yeah I indented with spaces because you can't use tab in freaking github comments for some reason. Now, the bad argument definition... don't know why you're getting that.

Reply:
/mob/living/simple_animal/hostile/guardian/beam/New() //Dies if the summoner dies
	..()
	if(summoner)
		if(!(NO_SHOCK in summoner.mutations))
			summoner.mutations.Add(NO_SHOCK)`